spec/expression.dd

```diff
@@ -1623,44 +1623,44 @@ $(GNAME AssertExpression):
     $(D assert $(LPAREN)) $(GLINK AssignExpression) $(D ,) $(GLINK AssignExpression) $(D ,)$(OPT) $(D $(RPAREN))
 )
 
-    $(P The assert expression is used to declare conditions that the
-        programmer asserts must hold at that point in the program if the
-        program logic has been correctly implemented. It can be used both as a
-        debugging tool and as a way of communicating to the compiler facts
-        about the code that it may employ to produce more efficient code.
+    $(P The first $(I AssignExpression) must evaluate to true. If it does not, an $(I Assert Failure)
+    has occurred and the program enters an $(I Invalid State).
     )
 
-    $(P Programs for which $(I AssignExpression) is false are invalid.
-        Subsequent to such a false result, the program is in an invalid,
-        non-recoverable state.
+    $(P The expression $(D assert(0)) is a special case; it
```

Comment:
We need to clarify this further. Are these also special-cased?

```d
assert(1 - 1);
assert(10 * 10 - 100 == 0);
enum a = 0;
assert(a);
enum b = sqrt(100);
assert(b == 10);
assert(false);
assert(null);
assert([]);
```

In other words, to what extent does the compiler go to evaluate the expression during compilation? I don't know the answer.